### PR TITLE
DBZ-5035 pass sink config props to db history adapter

### PR DESCRIPTION
**Pull request by zalmane**

### CODE REVIEW

**Review by jpechane — CHANGES_REQUESTED** [review]

File: debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseConnectorConfig.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public DatabaseHistory getDatabaseHistory() {
         }
 
         // Do not remove the prefix from the subset of config properties ...
-        Configuration dbHistoryConfig = config.subset(DatabaseHistory.CONFIGURATION_FIELD_PREFIX_STRING, false)
+        Configuration dbHistoryConfig = config.filter(k -> k.startsWith(DatabaseHistory.CONFIGURATION_FIELD_PREFIX_STRING) || k.startsWith("debezium.sink."))
```
[the remark this answers is not in the document]
Comment (zalmane) [reply debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseConnectorConfig.java:104]: will this be OK in terms of dependencies when packaging this outside of Debezium server?

[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseConnectorConfig.java:104]: Damn you are right. Unfortunately the fix will not be that easy then as server refs (even indirect) should not leak into the core.

**Comment by zalmane** [comment]
do you mean regarding referencing the variable or in general? I can add a
constant at the class level.
another option is to pass the entire config to the history adapter instead
of limiting it to the history related settings.
WDYT?

File: debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseConnectorConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (zalmane) [reply debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseConnectorConfig.java:104]: do you mean regarding referencing the variable or in general? I can add another local
constant at the class level.
Another option is to pass the entire config to the history adapter instead
of limiting it to the history related settings, which may be useful for other purposes.
WDYT?

[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseConnectorConfig.java:104]: @zalmane Could you please take a look at this one so we can wrap it up? Thanks

[the remark this answers is not in the document]
Comment (ggaborg) [reply debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseConnectorConfig.java:104]: Hi, check out this new pull request as per your recommendations: https://github.com/debezium/debezium/pull/3797

**Comment by jpechane** [comment]
Implemented via https://github.com/debezium/debezium/pull/3797

**Pull request closed without merging**